src/framework/theme/components/layout/direction.directive.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import { ChangeDetectorRef, Directive, OnDestroy, OnInit, TemplateRef, ViewContainerRef } from '@angular/core';
+import { NbLayoutDirection, NbLayoutDirectionService } from '../../services/direction.service';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+
+/**
+ * There are two structural directives which display the template content depending on layout direction.
+ * `*nbLtr` displays the content when layout direction is `ltr`.
+ * `*nbRtl` displays the content when layout direction is `rtl`.
+ * Use them without any conditions.
+ */
+@Directive()
+export abstract class NbLayoutDirectionDirective implements OnInit, OnDestroy {
+  private destroy$ = new Subject<void>();
+
+  private hasView = false;
+
+  constructor(
+    protected templateRef: TemplateRef<any>,
+    protected viewContainer: ViewContainerRef,
+    protected cd: ChangeDetectorRef,
+    protected directionService: NbLayoutDirectionService,
+    protected direction: NbLayoutDirection,
+  ) {}
+
+  ngOnInit() {
+    this.directionService
+      .onDirectionChange()
+      .pipe(takeUntil(this.destroy$))
```

Comment:
Add `distinctUntilChanged` operator as direction could be set to the same value multiple times.